torch/_meta_registrations.py

```diff
@@ -7011,6 +7011,60 @@ def meta_local_scalar_dense(self: Tensor):
     raise RuntimeError("Tensor.item() cannot be called on meta tensors")
 
 
+@register_meta(aten.silu)
+@out_wrapper()
+def silu(self: Tensor) -> Tensor:
```

Comment:
Dtype promotion of out= functions on meta inputs shoule be consistent with CPU, refer to [link](https://github.com/pytorch/pytorch/issues/138399) for more information

```suggestion
@out_wrapper(exact_dtype=True)
def silu(self: Tensor) -> Tensor:
```